### PR TITLE
fedora-coreos-config-transpiler: init at 0.6.0

### PR DESCRIPTION
**Pull request by ruuda**
###### Motivation for this change

We package the CoreOS [Container Linux Config Transpiler](https://github.com/coreos/container-linux-config-transpiler), but CoreOS Container Linux [has reached end of life](https://coreos.com/os/eol/), and has been superseded by Fedora CoreOS. The [Fedora CoreOS Config Transpiler](https://github.com/coreos/fcct) (FCCT) is the equivalent config transpiler for Fedora CoreOS.

I am not removing the original Container Linux Config Transpiler, because Container Linux lives on in [Flatcar Linux](https://www.flatcar-linux.org/).

The new package is a copy of the container-linux-config-transpiler package, with metadata updated and `postInstall` slightly simplified.

@pop I added you as a maintainer because you are the maintainer of the original config transpiler package, are you okay with that?

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by flokli — COMMENTED** [review]
A suggestion, and some style nitpicks (probably just copypasta from `pkgs/development/tools/container-linux-config-transpiler/default.nix`.

File: pkgs/development/tools/fedora-coreos-config-transpiler/default.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+  postInstall = ''
+    mv $out/bin/{internal,fcct}
+  '';
+
+  meta = {
+    description = "Translates Fedora CoreOS configs into Ignition configs";
+    license = licenses.asl20;
+    homepage = "https://github.com/coreos/fcct";
+    maintainers = with maintainers; [elijahcaine ruuda];
```
[the remark this answers is not in the document]
Comment (ruuda) [reply pkgs/development/tools/fedora-coreos-config-transpiler/default.nix]: Thanks, applied.

File: pkgs/development/tools/fedora-coreos-config-transpiler/default.nix [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    platforms = with platforms; unix;
```
[the remark this answers is not in the document]
Comment (ruuda) [reply pkgs/development/tools/fedora-coreos-config-transpiler/default.nix]: Thanks, applied.

File: pkgs/development/tools/fedora-coreos-config-transpiler/default.nix [hunk 3]
```diff
@@ -0,0 +1,34 @@
+{ lib, fetchFromGitHub, buildGoPackage }:
+
+with lib;
+
+buildGoPackage rec {
```
[the remark this answers is not in the document]
Comment (ruuda) [reply pkgs/development/tools/fedora-coreos-config-transpiler/default.nix]: I did now, it’s in 05c559cbc9b8f63ab4286c1b3a83b00ccbfeec90. I did stumble upon #84826, but it is not clear to me what the conclusion from that was. Is `buildGoModule` preferred in this case?

[the remark this answers is not in the document]
Comment (flokli) [reply pkgs/development/tools/fedora-coreos-config-transpiler/default.nix]: Yeah, reasoning is in https://github.com/NixOS/nixpkgs/pull/86376. Thanks!

**Pull request merged**